Sources/GithubData.swift

```diff
@@ -0,0 +1,205 @@
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import PerfectLib
+import PerfectLogger
+
+public class GithubData : JSONConvertibleObject, CustomStringConvertible {
+  static let registerName = "githubData"
+
+  var action: String = ""
+  var PRData: PullRequestData?
+  var issueData: IssueData?
+  public var description: String {
+    return "GithubData: \(action), \(PRData?.description ?? "No PR Data"), \(issueData?.description ?? "No Issue Data")"
+  }
+
+  init(action: String, PRData: [String: Any]?, issueData: [String: Any]?) {
+    self.action = action
+    if let PRData = PRData {
+      self.PRData = PullRequestData.createPRData(from: PRData)
+    }
+    if let issueData = issueData {
+      self.issueData = IssueData.createIssueData(from: issueData)
+    }
+  }
+
+  public override func setJSONValues(_ values: [String : Any]) {
+    self.action = getJSONValue(named: "action", from: values, defaultValue: "")
+    self.PRData = getJSONValue(named: "pull_request", from: values, defaultValue: nil)
+    self.issueData = getJSONValue(named: "issue", from: values, defaultValue: nil)
+  }
+
+  public override func getJSONValues() -> [String : Any] {
+    return ["action": action,
+            "pull_request": PRData as Any,
+            "issue": issueData as Any]
+  }
+
+  class func createGithubData(from json: String) -> GithubData? {
+    do {
+      guard let incoming = try json.jsonDecode() as? [String: Any] else {
+        return nil
+      }
+      return GithubData(action: incoming["action"] as? String ?? "",
+                        PRData: incoming["pull_request"] as? [String: Any] ?? nil,
+                        issueData: incoming["issue"] as? [String: Any] ?? nil)
+    } catch {
+      return nil
+    }
+  }
+
+}
+
+public class PullRequestData: JSONConvertibleObject, CustomStringConvertible {
+  static let registerName = "pullRequestData"
+
+  var id: Int = -1
+  var html_url: String = ""
+  var diff_url: String = ""
+  var state: String = ""
+  var title: String = ""
+  var body: String = ""
+  var labels: [String] = [String]()
+  var url: String = ""
+  var issue_url: String = ""
+  public var description: String {
+    return "PullRequestData: id:\(id), title:\(title), body:\(body), state:\(state), diff_url:\(diff_url)"
+  }
+
+  init(id: Int,
+       html_url: String,
+       diff_url: String,
+       state: String,
+       title: String,
+       body: String,
+       labels: [String],
+       url: String,
+       issue_url: String) {
+    self.id = id
+    self.html_url = html_url
+    self.diff_url = diff_url
+    self.state = state
+    self.title = title
+    self.body = body
+    self.labels = labels
+    self.url = url
+    self.issue_url = issue_url
+  }
+
+  public override func setJSONValues(_ values: [String : Any]) {
```

Comment:
Can these be properties?  (Here and anywhere else we have a set/get pair)

Reply:
unfortunately we are overriding specific functions, please see: https://github.com/PerfectlySoft/Perfect/blob/master/Sources/PerfectLib/JSONConvertible.swift#L77